tests/parsers/test.properties.js

```diff
@@ -0,0 +1,53 @@
+import Linter from 'linter';
+import PropertiesParser from 'parsers/properties';
+
+describe('PropertiesParser', () => {
+  it('should parse valid .properties file correctly', () => {
+    const addonLinter = new Linter({ _: ['bar'] });
+    const propertiesParser = new PropertiesParser(`
+      foo=bar
+      abc.def = xyz
+      #Comment
+      fast=ball
+
+      #Empty spaces
+
+      this.is.a =test
+      overwrite=foo
+      overwrite=bar`, addonLinter.collector);
+
+    propertiesParser.parse();
+
+    expect(propertiesParser.isValid).toEqual(true);
+    expect(propertiesParser.parsedData).toEqual({
+      foo: 'bar',
+      'abc.def': 'xyz',
+      fast: 'ball',
+      'this.is.a': 'test',
+      overwrite: 'bar',
+    });
+  });
+
+  it('should ignore invalid entities', () => {
+    const addonLinter = new Linter({ _: ['bar'] });
+    const propertiesParser = new PropertiesParser(`
+        this should be ignored.
+        foo=
+        bar
+        abc.def=x
+        yz
+        three.lines=a
+        b
+        c
+        d`, addonLinter.collector);
+
+    propertiesParser.parse();
+
+    expect(propertiesParser.isValid).toEqual(true);
+    expect(propertiesParser.parsedData).toEqual({
+      foo: 'bar',
+      'abc.def': 'xyz',
+      'three.lines': 'abcd',
```

Comment:
Do we care what the values are? I'm guessing not but I'd imagine you'd want a space between the lines.

Reply:
I don't think we do, at least not right now and I'm sure if we would we mostly care about the keys (e.g for completeness checks).